opt: pkg/sql/opt/exec/execbuilder test broken in the bazel sandbox

The failure looks like this at `HEAD` on my machine:

```
        --- FAIL: TestExecBuild/local/show_trace_nonmetamorphic (1.82s)
            --- FAIL: TestExecBuild/local/show_trace_nonmetamorphic/autocommit (0.02s)
```

As before, the goal here is to remove the `broken_in_bazel` tag from the test.

--- a/pkg/sql/opt/exec/execbuilder/BUILD.bazel
+++ b/pkg/sql/opt/exec/execbuilder/BUILD.bazel
@@ -47,7 +47,7 @@ go_library(
 
 go_test(
     name = "execbuilder_test",
-    size = "small",
+    size = "medium",
     srcs = [
         "builder_test.go",
         "main_test.go",
@@ -55,7 +55,6 @@ go_test(
     data = glob(["testdata/**"]) + [
         "@cockroach//c-deps:libgeos",
     ],
-    tags = ["broken_in_bazel"],
     deps = [
         "//pkg/security",
         "//pkg/security/securitytest",

